Feature/data table

sejpat25 opened the pull request:
This is a feature for the dashboard intending to be below the graph. It consists of a drop-down menu where each element is a card (that corresponds to a node). These cards are expandable so upon a user clicking on the arrow icon it should open or close depending on the state. 

There are still things that need to be completed or taken into account for this feature including:
- The input of nodes and related information obtained from other components of the application.
- The formatting (width and minimum height for a scrollbar to be added).

sejpat25 committed:
Installed library associated with expandable panels

Includes code for the component to display this information (manually inserted).

sejpat25 committed:
Cards can be rendered by extracting relevant info from array

Instead of having replicated code for each individual card, each card  renders information according to the current element of the array. So, a template was created for each card.

sejpat25 committed:
Implemented Cards to be displayed in a drop down menu

Each individual node card will be displayed in the drop down menu. Once it is clicked the card will expand displaying new information. Also removed a now unnecessary file.

sejpat25 committed:
Completed a Code Tidy Up

Renaming classes, variables and removing unnecessary code. Formatted the 'dummy' data to give a better idea of what a card will look like.

sejpat25 committed:
Configured props for node information to move between components

Information (with manual input at this stage) can be transferred from the dashboard component to the drop down node list using props. The manual input needs to be updated to the nodes that the user selects.

sejpat25 committed:
Fixed the code regarding a warning that appeared

There was a red warning that appeared with iterating through the props. Have fixed this error.

sejpat25 committed:
Merge branch 'develop' of https://github.com/Edds390/solarnetwork-aggregated-dashboard into feature/data-table

sejpat25 committed:
Merge branch 'develop' of https://github.com/Edds390/solarnetwork-aggregated-dashboard into feature/data-table

sejpat25 committed:
Commented code and adjusted some prop configurations

Placed the drop down menu on the same dashboard component as the chart. Have also made some minor adjustments with respect to demonstrating an example of props and data movement between components.

Edds390 commented:
Good commenting.
Eslint is picking up a few issues. Make sure you have the eslint plugin installed in vs-code.
![image](https://user-images.githubusercontent.com/21207045/37873213-0f8afe24-3074-11e8-8c41-e2ef19a55a01.png)

Also components should have separate directories so you'd want to have DropDownNodeMenu/DropDownNodeMenu.js and ExpandableCard/ExpandableCard.js

sejpat25 committed:
Made adjustments according to the feedback from my pull request

Used eslint to fix formatting issues and changed the directory structure for the components I was working on.

sejpat25 committed:
Fixed some more eslint errors but still need fix the props error

sejpat25 committed:
Fixed the prop types mapping eslint error

SeoyoungC commented on src/components/DataTable/DropDownNodeMenu/DropDownNodeMenu.js:
```diff
@@ -0,0 +1,65 @@
+const nodeMap = [
+  {
+    title: 'Node 263',
+    minPower: 'Min Power: 288 Watts',
```
Although just dummy data, could change the map to only store integer values and have the components themselves fill in the text, e.g. 'minPower: 288', and have the component paint the 'Min Power: ' text.
Would reduce the amount of duplication - just something to keep in mind when bringing in real data!

sejpat25 committed:
Changed input values in the array to integers only

When we extract information from the API calls, chances are it will be only numerical values. So updated the formatting to accomodate this.

SeoyoungC reviewed (approved):
Looks good to me! 👍
